mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala

```diff
@@ -40,9 +40,10 @@ private[spark] object SchemaUtils {
       msg: String = ""): Unit = {
     val actualDataType = schema(colName).dataType
     val message = if (msg != null && msg.trim.length > 0) " " + msg else ""
-    require(actualDataType.equals(dataType),
-      s"Column $colName must be of type ${dataType.catalogString} but was actually " +
-        s"${actualDataType.catalogString}.$message")
+    require(actualDataType.getClass.equals(dataType.getClass) &&
```

Comment:
I mean I don't think you need this extra check; it already fails. I'm also not sure if this ends up being overly restrictive in some other cases.
It's the message that's not helpful. The message update is sound.

Reply:
I agree with you, class check may be too strict, I will try to find other more useful way if I still have the chance.

Reply:
Thanks once more,my former behavior seems a little bad.

Reply:
` require(actualDataType.equals(dataType),
      s"Column $colName must be of type ${dataType.getClass}:${dataType.catalogString} " +
        s"but was actually ${actualDataType.getClass}:${actualDataType.catalogString}.$message")`

Reply:
> ` require(actualDataType.equals(dataType), s"Column $colName must be of type ${dataType.getClass}:${dataType.catalogString} " + s"but was actually ${actualDataType.getClass}:${actualDataType.catalogString}.$message")`Current version is just do some thing right in my opinion.